### PR TITLE
Migrate AvatarToken

### LINKED ISSUE #18885: Migrate components to TS: AvatarToken
### Description
Migrate `AvatarToken` component to TypeScript

### Technical Details
- Convert Objects in `SNAKE_CASE` to enums in `PascalCase` follow [TS conventions](https://www.notion.so/Use-TypeScript-9ec7bd8fc8cf4d82b81cd8943c4707cd?pvs=4#7e6410b21100455a9a08442108f19d78)
- Update file name from `component-name.contants.js` to `component-name.types.ts` and add types
- Update file name from `component-name.js` to `component-name.tsx` and add types
- Update file name from `component-name.test.js` to `component-name.test.tsx`
- Update file name from `component-name.stories.js` to `component-name.stories.tsx`
- Update all instances of component objects to enums
- Update all documentation 

### Acceptance Criteria
- Component has been converted to TypeScript

### PR DESCRIPTION
**Pull request by thebinij**
## Explanation

* Closes #18885 

## Pre-merge author checklist

- [X] I've clearly explained:
  - [X] What problem this PR is solving
- [ ] Sufficient automated test coverage has been added

## Pre-merge reviewer checklist

- [ ] Manual testing (e.g. pull and build branch, run in browser, test code being changed)
- [X] PR is linked to the appropriate GitHub issue
- [ ] **IF** this PR fixes a bug in the release milestone, add this PR to the release milestone

If further QA is required (e.g. new feature, complex testing steps, large refactor), add the `Extension QA Board` label.

In this case, a QA Engineer approval will be be required.

### CODE REVIEW

File: shared/constants/network.ts [hunk 1]
```diff
@@ -3,28 +3,28 @@ import { capitalize, pick } from 'lodash';
...
 /**
  * Test networks have special symbols that combine the network name and 'ETH'
  * so that they are distinct from mainnet and other networks that use 'ETH'.
  */
 export type TestNetworkCurrencySymbol =
-  (typeof TEST_NETWORK_TICKER_MAP)[keyof typeof TEST_NETWORK_TICKER_MAP];
+  typeof TEST_NETWORK_TICKER_MAP[keyof typeof TEST_NETWORK_TICKER_MAP];
 
 /**
  * An object containing preferences for an RPC definition
```
Comment (georgewrmarshall) [inline shared/constants/network.ts:30]: I don't think this file should have changes even if just formatting. Could you please revert thanks!

File: ui/components/component-library/avatar-token/avatar-token.stories.tsx [hunk 2]
```diff
@@ -324,7 +323,9 @@ ShowHalo.args = {
   showHalo: true,
 };
 
-export const ColorBackgroundColorAndBorderColor = (args) => (
+export const ColorBackgroundColorAndBorderColor: ComponentStory<
+  typeof AvatarToken
+> = (args) => (
```
Comment (georgewrmarshall) [inline ui/components/component-library/avatar-token/avatar-token.stories.tsx:328]: nit: does this need to be on a new line or was it automatically formatted that way by eslint?
```suggestion
export const ColorBackgroundColorAndBorderColor: ComponentStory<typeof AvatarToken> = (args) => (
```

File: ui/components/component-library/avatar-token/avatar-token.types.ts [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+  /**
+   * The showHalo accepts a boolean prop to render the image with halo effect
+   */
+  showHalo?: boolean;
+  /**
+   * The size of the AvatarToken.
+   * Possible values could be 'AvatarBaseSize.Xs' 16px, 'AvatarBaseSize.Sm' 24px, 'AvatarBaseSize.Md' 32px, 'AvatarBaseSize.Lg' 40px, 'AvatarBaseSize.Xl' 48px
+   * Defaults to AvatarBaseSize.Md
+   */
+  size?: AvatarBaseSize;
```
Comment (georgewrmarshall) [inline ui/components/component-library/avatar-token/avatar-token.types.ts:31]: Same question as AvatarNetwork is there a way we could allow for AvatarNetworkSize enum? Not sure if we want to mix enum names and components? Do you have any thoughts on this @brad-decker? If we update the engineer will have to use something like

```
<AvatarToken size={AvatarBaseSize.Lg} />
```

Which I'm not a huge fan of...

File: ui/components/ui/box/box.d.ts [hunk 4]
```diff
@@ -290,7 +290,7 @@ export interface BoxProps extends React.HTMLAttributes<HTMLElement> {
    * Use TEXT_ALIGN const from '../../../helpers/constants/design-system';
    * Accepts responsive props in the form of an array.
    */
-  textAlign?: BoxTextAlign | BoxTextAlignArray;
+  textAlign?: BoxTextAlign | BoxTextAlignArray | TextAlign;
```
Comment (georgewrmarshall) [inline ui/components/ui/box/box.d.ts:293]: Left a comment in another of your PRs addressing this

**Review by georgewrmarshall — COMMENTED** [review]
Hey @thebinij, this is looking good would you be able to add some before/after screencasts or screenshots to ensure no visual regressions before and after the migration? Thanks

File: shared/constants/network.ts (hunk 1 above)
Comment (thebinij) [reply shared/constants/network.ts:30]: Sure!

File: ui/components/ui/box/box.d.ts (hunk 4 above)
Comment (thebinij) [reply ui/components/ui/box/box.d.ts:293]: Got it!

File: ui/components/component-library/avatar-token/avatar-token.stories.tsx (hunk 2 above)
Comment (thebinij) [reply ui/components/component-library/avatar-token/avatar-token.stories.tsx:328]: It was automatically formatted by lint:eslint:fix

**Review by Karlitos88 — APPROVED** [review]

**Comment by thebinij** [comment]
This is an old PR. [New PR](https://github.com/MetaMask/metamask-extension/pull/19080)

**Pull request closed without merging**